### PR TITLE
Add docs to public facing functions

### PR DESCRIPTION
**Pull request by KonradStaniec**
Add docs to some of the public facing functions

### CODE REVIEW

File: btcstaking/types.go [hunk 1]
```diff
@@ -456,6 +462,12 @@ type UnbondingInfo struct {
 	slashingPathLeafHash chainhash.Hash
 }
 
+// BuildUnbondingInfo builds all Babylon specific BTC scripts that must
+// be committed to in the unbonding output.
+// Returned `StakingInfo` object exposes methods to build spend info for each
```
Comment (gitferry) [inline btcstaking/types.go:467]:
```suggestion
// Returned `UnbondingInfo` object exposes methods to build spend info for each
```

File: btcstaking/witness_utils.go [hunk 2]
```diff
@@ -17,7 +17,15 @@ func (si *SpendInfo) CreateTimeLockPathWitness(delegatorSig *schnorr.Signature)
 	return CreateWitness(si, [][]byte{delegatorSig.Serialize()})
 }
 
-func (si *SpendInfo) CreateUnbondingPathWitness(covenantSigs []*schnorr.Signature, delegatorSig *schnorr.Signature) (wire.TxWitness, error) {
+// CreateUnbondingPathWitness helper function to create a witness to spend
+// transaction through the unbonding path.
+// It is up to the caller to ensure that the amount of covenantSigs matches the
+// expected quorum of covenenant members or that transaction is has unbonding
+// path.
```
Comment (gitferry) [inline btcstaking/witness_utils.go:24]:
```suggestion
// It is up to the caller to ensure that the amount of covenantSigs matches the
// expected quorum of covenenant members and the transaction has unbonding path.
```

File: btcstaking/witness_utils.go [hunk 3]
```diff
@@ -46,7 +54,16 @@ func (si *SpendInfo) CreateUnbondingPathWitness(covenantSigs []*schnorr.Signatur
 	return CreateWitness(si, witnessStack)
 }
 
-func (si *SpendInfo) CreateSlashingPathWitness(covenantSigs []*schnorr.Signature, fpSigs []*schnorr.Signature, delegatorSig *schnorr.Signature) (wire.TxWitness, error) {
+// CreateSlashingPathWitness helper function to create a witness to spend
+// transaction through the slashing path.
+// It is up to the caller to ensure that the amount of covenantSigs matches the
+// expected quorum of covenenant members or that transaction is has slashing
+// path.
```
Comment (gitferry) [inline btcstaking/witness_utils.go:61]:
```suggestion
// expected quorum of covenenant members, the finality provider sigs respect the finality providers
// that the delegation belongs to, and the transaction has slashing path.
```

**Review by gitferry — APPROVED** [review]
Some nits:

**Review by RafilxTenfen — APPROVED** [review]

**Pull request merged**